### ISSUE TITLE
Websockets support

### ISSUE THREAD

**Issue by oskarth** [status-im/nim-libp2p#407]
# Problem

For Waku v2 to run in a web browser, an important design goal, there needs to be a way to communicate between the browser and a Waku node.

The current PoC (https://github.com/vacp2p/waku-web-chat/) is using a basic NodeJS implementation supporting TCP, Websockets and WebRtcStar. This then acts as a bridge between browser (WS/WRTC) and nim-waku (TCP) and relays messages between them.

However, it'd be useful to replace this with nim-waku. This will be needed to avoid duplicate work, and other protocols are req-resp, which would mean creating a full implementation of Waku v2 in NodeJS, which is obviously not desirable if we can avoid it.

# Acceptance criteria

- Websockets transport support
- Can use to support e.g. Waku Web browser
- Works in a deployed and secure environment (wss)

**Comment by oskarth** [comment]
What's the state of this @dryajov? This is currently blocking js-waku browser roadmap and means we have to duplicate efforts by porting nim-waku functionality to go-waku.

**Comment by oskarth** [comment]
@dryajov what is the state of this?

**Comment by Menduist** [comment]
It's being worked on in #593, it should work for a basic use, but `wss` hasn't been tested (and `nim-websock` would also need more tests for prod use)

It's currently blocked by #598, which is itself blocked by chronos
But everything should unlock in the coming weeks :)

**Comment by dryajov** [comment]
> wss hasn't been tested

Should be fully supported by nim-websock

> and nim-websock would also need more tests for prod use

We're testing this with autobahn - https://github.com/crossbario/autobahn-testsuite, which is used by many big projects and established products, including mozilla's firefox. We currently have 100% coverage when it comes to implementing the RFC, where you might find issues is with corner cases or simple oversights.

**Comment by D4nte** [comment]
The setup we are using in the mean time (websockify) is not really stable. Can we have a status on this one please? I see #598 is still open.

**Comment by Menduist** [comment]
It has been merged into unstable #593

**Comment by Menduist** [comment]
Closed in #593
